test/generator/expected_output/compiled_output.coffee

```diff
@@ -186,34 +186,35 @@ class CompiledOutput
                 Math.sqrt( 100 );
                 Math.sqrt( 9 );
                 Math.sqrt( 1 );
-                Math.sqrt( 0 );
+                return Math.sqrt( 0 );
             }());
         }());
     '''
+
   @program14 =
     '''(function () {
           (function () {
-              ( (function(lb, ub, skip) {
-                  var temp = [];
-                  for(var i = lb; i < ub; i += skip ) {
-                      temp.push(i);
+              ( (function (lb, ub, skip) {
+                  var buffer = [];
+                  for ( var i = lb; i < ub; i += skip ) {
+                      buffer.push( i );
                   }
-                  return temp;
-              })( 1, 10 - 1, 1)).forEach( function (_v1) {
-                              (function () {
-                      ( (function(lb, ub, skip) {
```

Comment:
Lots of win! This one was particularly difficult to track down. If I had to do it over again, I'd probably do our emit differently because it's bug prone, but given it's so late in the semester, and we need to get this done, I'm happy leaving it the way it is.